lib/parallel-write.js

```diff
@@ -42,3 +52,65 @@ function writableIsFull(s) {
 
     return state.length >= state.highWaterMark;
 }
+
+/** sentry is a magical special best.
+
+We must deep copy the log entry because sentry mutates it.
+This means if we write to sentry first we will write malformed
+data to kafka or disk.
+
+This is a magical deep copy because it can deep copy error objects
+
+*/
+function magicCopy(entry) {
+    if (isError(entry)) {
+        return magicCopyError(entry);
+    }
+
+    var copy = deepExtend({}, entry);
+
+    if (typeof entry === 'object' && entry !== null) {
+        Object.keys(entry).forEach(function copyProp(k) {
+            var value = entry[k];
+            if (isError(value)) {
```

Comment:
Probably wise to make a stack [] and not recur for hot code. Also, isError in two places in a recursive function can be simplified.